### PR TITLE
fixed firewall: firewall_policy_key not considered

### PR DESCRIPTION
**Pull request by paullly**
# [Issue-id](https://github.com/aztfmod/terraform-azurerm-caf/issues/ISSUE-ID-GOES-HERE)

## PR Checklist

---

- [ ] I have added example(s) inside the [./examples/] folder
- [ ] I have added the example(s) to the integration test list for [normal (~30 minutes)](./workflows/standalone-scenarios.json) or [long runner >30 minutes](./workflows/standalone-scenarios-longrunners.json)
- [x] I have checked the [coding conventions as per the wiki](https://github.com/aztfmod/terraform-azurerm-caf/wiki)
- [x] I have checked to ensure there aren't other open Pull Requests for the same update/change?

## Description

The firewall_policy_key was not considered when creating an azure firewall

## Does this introduce a breaking change

- [ ] YES
- [x ] NO

### CODE REVIEW

File: networking_firewall.tf [hunk 1]
```diff
@@ -24,7 +24,7 @@ module "azurerm_firewalls" {
   virtual_networks    = local.combined_objects_networking
   virtual_wans        = local.combined_objects_virtual_wans
 
-  firewall_policy_id = can(each.value.firewall_policy.id) || can(each.value.firewall_policy) == false ? try(each.value.firewall_policy.id, null) : local.combined_objects_azurerm_firewall_policies[try(each.value.firewall_policy.lz_key, local.client_config.landingzone_key)][try(each.value.firewall_policy.key, each.value.firewall_policy_key)].id
+  firewall_policy_id = can(each.value.firewall_policy.id) || can(each.value.firewall_policy) ? try(each.value.firewall_policy.id, null) : local.combined_objects_azurerm_firewall_policies[try(each.value.firewall_policy.lz_key, local.client_config.landingzone_key)][try(each.value.firewall_policy.key, each.value.firewall_policy_key)].id
```
Comment (LaurentLesle) [inline networking_firewall.tf:27]:
```suggestion
  firewall_policy_id = can(each.value.firewall_policy.id) ? each.value.firewall_policy.id : try(local.combined_objects_azurerm_firewall_policies[try(each.value.firewall_policy.lz_key, local.client_config.landingzone_key)][try(each.value.firewall_policy.key, each.value.firewall_policy_key)].id, null)
```

Comment (LaurentLesle) [reply networking_firewall.tf:27]: Thanks for submitting the PR. Indeed a regression. However can you consider the code above?

**Review by LaurentLesle — CHANGES_REQUESTED** [review]

File: networking_firewall.tf [hunk 2]
```diff
@@ -24,7 +24,7 @@ module "azurerm_firewalls" {
   virtual_networks    = local.combined_objects_networking
   virtual_wans        = local.combined_objects_virtual_wans
 
-  firewall_policy_id = can(each.value.firewall_policy.id) || can(each.value.firewall_policy) == false ? try(each.value.firewall_policy.id, null) : local.combined_objects_azurerm_firewall_policies[try(each.value.firewall_policy.lz_key, local.client_config.landingzone_key)][try(each.value.firewall_policy.key, each.value.firewall_policy_key)].id
+  firewall_policy_id = can(each.value.firewall_policy.id) ? each.value.firewall_policy.id : try(local.combined_objects_azurerm_firewall_policies[try(each.value.firewall_policy.lz_key, local.client_config.landingzone_key)][try(each.value.firewall_policy.key, each.value.firewall_policy_key)].id, null)
```
Comment (LaurentLesle) [inline networking_firewall.tf]:
```suggestion
  firewall_policy_id = can(each.value.firewall_policy.id) || ( can(each.value.firewall_policy) && can(each.value.firewall_policy.key) == false ) ? try(each.value.firewall_policy.id, null) : local.combined_objects_azurerm_firewall_policies[try(each.value.firewall_policy.lz_key, local.client_config.landingzone_key)][try(each.value.firewall_policy.key, each.value.firewall_policy_key)].id
```
```suggestion
  firewall_policy_id = can(each.value.firewall_policy.id) ? each.value.firewall_policy.id : try(local.combined_objects_azurerm_firewall_policies[try(each.value.firewall_policy.lz_key, local.client_config.landingzone_key)][try(each.value.firewall_policy.key, each.value.firewall_policy_key)].id, null)
```

**Review by LaurentLesle — CHANGES_REQUESTED** [review]

**Review by LaurentLesle — APPROVED** [review]
LGTM

**Comment by arnaudlh** [comment]
good catch, thanks for your contribution @paullly!

**Pull request merged**